binary_transparency/firmware/cmd/ft_personality/internal/trees/trees.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package trees contains the personality tree configuration.
+package trees
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/crypto/keyspb"
+	"github.com/google/trillian/crypto/sigpb"
+	"google.golang.org/grpc"
+)
+
+// TreeStorage allows access to the configuration.
+type TreeStorage struct {
+	db *sql.DB
+}
+
+// NewTreeStorage sets up a TreeStorage using the given DB.
+func NewTreeStorage(db *sql.DB) *TreeStorage {
+	return &TreeStorage{
+		db: db,
+	}
+}
+
+// EnsureTree gets the tree for this personality, creating it if necessary.
+// Takes a connection to Trillian to use for initialization.
+func (s *TreeStorage) EnsureTree(ctx context.Context, conn grpc.ClientConnInterface) (*trillian.Tree, error) {
```

Comment:
Worth a comment here about the raciness of this?

Reply:
Done.